lua/luasnip/util/util.lua

```diff
@@ -185,7 +185,8 @@ local function get_selection()
 end
 
 local function store_selection()
-	local chunks = vim.split(vim.fn.getreg('"'), "\n")
+	local c = require("luasnip.config").config
+	local chunks = vim.fn.getreg(c.store_selection_register, 1, true)
```

Comment:
I did some quick testing, and it seems like this way, the unnamed register will still contain the visual. Setting it here with `setreg()` and the content of `"1` should make it work as expected.

Reply:
I tested it myself and I don't see it cluttering `"`.

Reply:
I mean, i did  `"ls`  by hand and it does not overrides `"`

Reply:
Oh, I tested it by just yanking text and then yanking some other text into `"l`, where it did overwrite unnamed.
But if it works, nvm that.

Reply:
Okay, I tested it on your branch, this seems to still overwrite the unnamed register :/

Reply:
`vim.fn.setreg('"', vim.fn.getreg("0"))` does the trick ;)

Reply:
If it overrides the register then the fix is a bit more complex, as that way you are recovering the content of the register, but not the type of it.

Reply:
Yeah, you're right, I looked into the docs, even `0` only works because I yanked the text.

Reply:
Was looking into this and there is not how to properly reset the unnamed register, as it points to the last named/numbered used register so setting, recovering from `"0` only works if the last action was a yank, if it was a delete then the content won't be there, but in `"1` or `"-`.  And now that I think about it, probably using the ultisnips dance won't fix it either as they do `s` too, so they override the unamed register the same way.

Reply:
We'd have to go back to the manual vim-dance and delete visual using nvim_buf_set_text(), that would leave registers alone.

Reply:
Another option would be to do it on the TextYankPost autocommand, with that there would be no expectation that the register remains untouched.

Reply:
So, I think there is not a proper way to fix this issue without overengineering (wrapping all the `s` operations to store all the registers and set them again after). So I think if someone wants to copy and paste in a selection snippet, they better use `"0p`, `"1` or `"-` accordingly (yes, that's painful)

Reply:
> We'd have to go back to the manual vim-dance and delete visual using `nvim_buf_set_text()`, that would leave registers alone.

That should work though, the only annoying part would be the complicated manual copying of the selection.
(it would also be possible to set the cursor as it is with `s`)

Reply:
One option could be to do the vim-dance and use  "_s  as that null register does not update the unnamed one.

Reply:
Ah yeah, that's better :+1: